packet_test.go

```diff
@@ -297,49 +459,49 @@ func TestSSHFxpOpenPackethasPflags(t *testing.T) {
 	}
 }
 
-func BenchmarkMarshalInit(b *testing.B) {
+func benchMarshal(b *testing.B, packet encoding.BinaryMarshaler) {
```

Comment:
Should we reuse the `writeToBuffer` here to also include copying the marshalled data into a buffer, while also not testing the `bytes.Buffer` object?

:thinking: I can see arguments either way… but I think we don’t want to. Copying the data out of the marshaled binary data is essentially irrelevant to the benchmarks we really want to be testing here.

In either case, we should ideally `b.ResetTimer()` here so as to not include the setup to the benchmark. (It’s a pretty trivial amount of work compared to the fast loop below, but principles.)

Reply:
So, the `copy(…)` command is going to take the same amount of time regardless of the implementation, even if split into two `bytes.Buffer.Write` calls. Adding that overhead into the benchmark actually overwhelms the time taken to prep the byte-slices to be written, which is really what we want to be benchmarking here.